File: app/src/main/java/com/example/chatapp/domain/data/People.kt
package com.example.chatapp.domain.data

data class People(
    var user: User,
    var isFriend: Boolean = false,
    var isRequestSent: Boolean = false,
    var isRequestReceived: Boolean = false
) {
    constructor() : this(User(), false, false, false)
}

sealed class FriendListItem {
    data class Header(val title: String) : FriendListItem()
    data class PersonItem(val people: People) : FriendListItem()
}

File: lib/entity/user/people.dart
import 'package:equatable/equatable.dart';
import 'package:flutter_study/entity/user/user.dart';

class People extends Equatable {
  final User user;
  final bool isFriend;
  final bool isRequestSent;
  final bool isRequestReceived;

  const People({
    required this.user,
    this.isFriend = false,
    this.isRequestSent = false,
    this.isRequestReceived = false,
  });

  const People.empty()
      : user = const User.empty(),
        isFriend = false,
        isRequestSent = false,
        isRequestReceived = false;

  @override
  List<Object?> get props => [
    user,
    isFriend,
    isRequestSent,
    isRequestReceived,
  ];
}

sealed class FriendListItem extends Equatable {
  const FriendListItem();

  @override
  List<Object?> get props => [];
}


class PeopleHeader extends FriendListItem {
  final String title;

  const PeopleHeader(this.title);

  @override
  List<Object?> get props => [title];
}

class PersonItem extends FriendListItem {
  final People people;

  const PersonItem(this.people);

  @override
  List<Object?> get props => [people];
}


